python/port/mod/turtle/modturtle.cpp

```diff
@@ -131,37 +133,63 @@ mp_obj_t modturtle_isdown() {
 }
 
 mp_obj_t modturtle_pencolor(size_t n_args, const mp_obj_t *args) {
-  if (n_args == 0) {
-    // pencolor()
-    KDColor c = sTurtle.color();
-    mp_obj_t mp_col[3];
-    mp_col[0] = mp_obj_new_int_from_uint(c.red());
-    mp_col[1] = mp_obj_new_int_from_uint(c.green());
-    mp_col[2] = mp_obj_new_int_from_uint(c.blue());
-    return mp_obj_new_tuple(3, mp_col);
+  switch(n_args){
+    case 0:{
+      // pencolor()
+      KDColor c = sTurtle.color();
+      mp_obj_t mp_col[3];
+      if(sTurtle.colorMode() == MicroPython::ColorParser::ColorModes::MaxIntensity255){
+        mp_col[0] = mp_obj_new_int_from_uint(c.red());
+        mp_col[1] = mp_obj_new_int_from_uint(c.green());
+        mp_col[2] = mp_obj_new_int_from_uint(c.blue());
+      } else {
+        mp_col[0] = mp_obj_new_float(c.red() / 255.0);
+        mp_col[1] = mp_obj_new_float(c.green() / 255.0);
+        mp_col[2] = mp_obj_new_float(c.blue() / 255.0);
+      }
+      return mp_obj_new_tuple(3, mp_col);
+    }
+    case 1:{
+      sTurtle.setColor(MicroPython::ColorParser::ParseColor(args[0], sTurtle.colorMode()));
+      break;
+    }
+    case 3:{
+      // pencolor(r, g, b)
```

Comment:
The following code is a duplicate of a part of `MicroPython::ColorParser::ParseColor`. We should create a tuple and call with a helper `ColorTuple(args[0], args[1],args[2])` (which takes the mode into account) and then call `MicroPython::ColorParser::ParseColor`.